packages/app/lib/qr-code/index.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { useMemo } from "react";
+
+import Svg, {
+  Defs,
+  G,
+  Path,
+  Rect,
+  Image,
+  ClipPath,
+  LinearGradient,
+  Stop,
+} from "react-native-svg";
+
+import { QRCodeProps } from "./type";
+import { genMatrix, transformMatrixIntoPath } from "./utils";
+
+const renderLogo = ({
+  size,
+  logo,
+  logoSize = 0,
+  logoBackgroundColor,
+  logoMargin = 0,
+  logoBorderRadius = 0,
+}: Pick<
+  QRCodeProps,
+  | "size"
+  | "logo"
+  | "logoSize"
+  | "logoBackgroundColor"
+  | "logoBorderRadius"
+  | "logoMargin"
+>) => {
+  const logoPosition = (size - logoSize - logoMargin * 2) / 2;
+  const logoBackgroundSize = logoSize + logoMargin * 2;
+  const logoBackgroundBorderRadius =
+    logoBorderRadius + (logoMargin / logoSize) * logoBorderRadius;
+
+  return (
+    <G x={logoPosition} y={logoPosition}>
+      <Defs>
+        <ClipPath id="clip-logo-background">
+          <Rect
+            width={logoBackgroundSize}
+            height={logoBackgroundSize}
+            rx={logoBackgroundBorderRadius}
+            ry={logoBackgroundBorderRadius}
+          />
+        </ClipPath>
+        <ClipPath id="clip-logo">
+          <Rect
+            width={logoSize}
+            height={logoSize}
+            rx={logoBorderRadius}
+            ry={logoBorderRadius}
+          />
+        </ClipPath>
+      </Defs>
+      <G>
+        <Rect
+          width={logoBackgroundSize}
+          height={logoBackgroundSize}
+          fill={logoBackgroundColor}
+          clipPath="url(#clip-logo-background)"
+        />
+      </G>
+      <G x={logoMargin} y={logoMargin}>
+        <Image
+          width={logoSize}
+          height={logoSize}
+          preserveAspectRatio="xMidYMid slice"
+          href={logo}
+          clipPath="url(#clip-logo)"
+        />
+      </G>
+    </G>
+  );
+};
+
+export const ReactQRCode = ({
```

Comment:
it forks from [react-native-qrcode-svg](https://github.com/awesomejerry/react-native-qrcode-svg) lib because I wanna try to custom QR code and make it look more beautiful. (but need to next time try)
also, it uses group `path` on the web, which will reduce the number of `dom` nodes even more, so I think it performs better than `react-qr-code`.